bgcval2/download_from_mass.py

```diff
@@ -418,59 +423,75 @@ def downloadMass(jobID, doMoo=True):
     fixFilePaths(outputFold, jobID)
     deleteBadLinksAndZeroSize(outputFold, jobID)
 
-    doLs = True
-    doDL = True
+    download_script_path = ''.join([folder('mass_scripts/'), jobID,'.sh'])
 
-    if not doMoo: return
-    if doLs:
-        print("Looking at the following files:")
-        ######
-        # print files
+    header_lines = ['# Run this script on mass-cli1.jasmin.ac.uk\n',]
+    header_lines.append('# from login1.jasmin.ac.uk, ssh to the mass machine:\n#     ssh -X  mass-cli\n')
+    header_lines.append(''.join(['# run script with:\n# source ', os.path.abspath(download_script_path),'\n']))
+    header_lines.append('# moo passwd -r # if mass password is expired\n')
+    download_script_txt = ''.join(header_lines)
+ 
+    #if not doMoo: return
+    print("download_from_mass:\tLooking at the following files:")
+    ######
+    # print files
+    #bashCommand = "moo passwd -r"
+    #print "running the command:",bashCommand
+    #process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+    #output = process.communicate()[0]
+    #print "output",output
 
-        #bashCommand = "moo passwd -r"
-        #print "running the command:",bashCommand
-        #process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
-        #output = process.communicate()[0]
-        #print "output",output
+    bashCommand = "moo ls moose:/crum/" + jobID + "/ony.nc.file/*.nc"
+    download_script_txt = ''.join([download_script_txt, bashCommand, '\n'])
 
-        bashCommand = "moo ls moose:/crum/" + jobID + "/ony.nc.file/*.nc"
-        print("running the command:", bashCommand)
+    if not doMoo:
+        print("download_from_mass:\tthe command is (dry-run): \n", bashCommand)
+        output = ''
+    else:
+        print("download_from_mass:\trunning the command:", bashCommand)
         stdout.flush()
-
         process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
         output = process.communicate()[0]
 
-        if len(output.split('\n')) > 6000:
-            #bashCommand = ''
-            failed = 0
-            process1 = {}
-            output1 = {}
-            #for i in reversed(range(100)):
-            for i in range(100):
-
-                bashCommand = "moo get --fill-gaps moose:/crum/" + jobID + "/ony.nc.file/*_1y_??" + mnStr(
-                    i) + "*.nc " + outputFold
-                print("running the command:", bashCommand)
-                stdout.flush()
-                try:
-                    process1[i] = subprocess.Popen(bashCommand.split(),
-                                                   stdout=subprocess.PIPE)
-                    process1[i].wait()
-                    output1[i] = process.communicate()[0]
-                    print(i, output1[i])
-                except:
-                    failed += 1
-                    print("Failed", i, '\t', bashCommand)
-            if failed > 10:
-                assert 0
+    if len(output.split('\n')) > 6000:
```

Comment:
what's 6000 lines so special?

Reply:
`moo get` can't handle huge numbers of files very well. UKESM produces 6 files per year, so 6000 lines is 1000 years of model  data. This splits the download into smaller more manageable parts for `moo get`.

Reply:
gotcha, cheers, bud! Just think if you want that to be a tunable parameter in the future